Add external switch configuration example for reference (#117)
harvester/docs: docs/networking/best-practice-vlan-aware-switch.md

```
switch# config terminal
switch(config)# interface ethernet1/<Port Number>
switch(config-if)# switchport
switch(config-if)# switchport mode trunk
switch(config-if)# switchport trunk allowed vlan 100-200
switch(config-if)# switchport trunk native vlan 100
switch(config-if)# no shutdown
switch(config-if)# end
switch# copy running-config startup-config
```

## Create a VLAN Network in Harvester

A new VLAN network can be created via the **Advanced > Networks** page and clicking the **Create** button.

Specify the name and VLAN ID that you want to create for the VLAN network <small>(You can specify the same vlan ID on different namespaces of [Rancher multi-tenancy](/rancher/virtualization-management/#multi-tenancy) support)</small>.
   ![create-vlan-network.png](assets/create-network.png)

### Connect VM to the same network of Harvester management network

Once users finished the configuration in the previous section, external switch will send out untagged network traffic for manangement. Inside Harvester, the default VLAN tag to receive untagged taffic is VLAN 1.

> If users need VM connects to VLAN 100, management network, users can create a VLAN Network in Harvester with VLAN ID 1 configuration instead of VLAN ID 100.

Author: VM only connect to VLAN 1.

we got some issue that asked about how to let VM access same VLAN of `harvester-mgmt` if harvester-mgmt is under VLAN 100 for example.

harvester VLAN 1 and external switch VLAN 100 are same subnet (same L2) in example architecture.
harvester will add VLAN 1 tag to untagged packet while recv untagged traffic. (outside to VM VLAN1)
harvester will remove VLAN 1 tag from VLAN 1 traffic while sending out VLAN 1 tag traffic  (VM VLAN1 to outside)

external switch will add VLAN 100 tag to untagged packet while recv untagged traffic (harvester to external switch)
external switch will remove VLAN 100 tag from any VLAN 100 traffic while sending out to harvester (external switch to harvester)

So if harvester-mgmt is under VLAN 100 of external switch.
harvester-mgmt didn't need to deal with VLAN tag because external switch already remove it.
but if VM wants to connect to the same VLAN of harvester-mgmt, harvester-br0 will add VLAN 1 tag to all untagged packet.
So VM need to connect to VLAN 1 in harvester as well to have same L2 domain with harvester-mgmt and VLAN 100 in external switch

> If users need VM connects to VLAN 100, management network, users can create a VLAN Network in Harvester with VLAN ID 1 configuration instead of VLAN ID 100.

Author: I hope i didn't confuse you about this.
mixed untagged and tagged VLAN confuse people a lot.
so i put another figure for explaination, hope this will help
![image](https://user-images.githubusercontent.com/2821179/153384279-7fb725b8-3aa5-4970-b00f-4d1bf448e772.png)

> External switch will remove VLAN 100 from the packet for egress and `harvester-br0` will add VLAN 1 to the packet and treat it as VLAN 1. Shown as the below diagram.

Reviewer: ```suggestion
The external switch will remove the VLAN 100 tag from the packet for egress and `harvester-br0` will add the VLAN 1 tag to the packet and treat it as VLAN 1 as shown in the following diagram:
```

   ![vlan-aware-native-vlan.png](assets/vlan-aware-native-vlan.png)

> !!! note
>     Do not create VLAN Network with VLAN 100 and associate any VM to it. Connectivity will not always be ensured and depend on external switch behavior to add/remove VLAN tag from packets.

Reviewer: ```suggestion
!!! warning
    Do not create a VLAN Network with VLAN 100 and associate any VM to it. The connectivity will not always be ensured and depends on the external switch behavior to add/remove VLAN tag from packets.
```

> ### Connect VM to Specific VLAN network

Reviewer: ```suggestion
### Connect a VM to specific VLAN network
```

> Users need to create VLAN Network with specific VLAN ID in need and asscicate VM to that VLAN network. Refer to [Harvester Network](/networking/harvester-network/)

Reviewer: ```suggestion
You need to create a VLAN Network with specific VLAN ID and associate the VM to that VLAN network. 

Please refer to [this page](/networking/harvester-network/) for additional information on Harvester Networking.
```

---
sidebar_position: 50
keywords:
  - Harvester
  - harvester
  - Rancher
  - rancher
  - Network
  - network
  - VLAN
  - vlan
Description: Harvester is built on top of Kubernetes, and uses the [CNI](https://github.com/containernetworking/cni) as the interface between network providers and Kubernetes pod networking. Naturally, we implement the Harvester network based on CNI. Moreover, the Harvester UI integrates the network configuration in order to provide a user-friendly way to configure networks for VMs.
---

# Best Practice for VLAN-aware Switch

In this best practice guide on how to configure "VLAN-aware", we will introduce Harvester VLAN network and external switch configuration for common scenario.

## Architecture

Hardware:
- Three Harvester servers with only one single port network card.
- One or more VLAN-aware switch(es). We will use "Cisco like" configuration as example.

Network Specification:

> - The Management Network to control Harvester servers needs to be set as VLAN 100.

Reviewer 2: Suggestion to change this sentence as:
Assume that the subnet of the Harvester hosts is in VLAN 100.

> - The Management Network to control Harvester servers needs to be set as VLAN 100.

Author: ```suggestion
- Assume that the subnet of the Harvester hosts is in VLAN 100.
```

> - The VM Network needs to be in the VLAN 101-200 range.

Reviewer 2: Suggestion to change this sentence as:
Assume that the VMs are in the VLAN 101-200.

> - The VM Network needs to be in the VLAN 101-200 range.

Author: ```suggestion
- Assume that the VMs are in the VLAN 101-200.
```

Cabling:

> - The Harvester servers are connected to the switch in a port from `1` to `3`.

Reviewer 3: ditto.

The following diagram illustrates the cabling used for this guide:

   ![vlan-aware-case.png](assets/vlan-aware-case.png)

## External Switch Configuration

For the external switch configuration, we'll use a "Cisco like" configuration as example. You can apply the following configurations to your switch:

```
switch# config terminal
switch(config)# interface ethernet1/<Port Number>
switch(config-if)# switchport
switch(config-if)# switchport mode trunk
switch(config-if)# switchport trunk allowed vlan 100-200
switch(config-if)# switchport trunk native vlan 100
switch(config-if)# no shutdown
switch(config-if)# end
switch# copy running-config startup-config
```

## Create a VLAN Network in Harvester

You can create a new VLAN network in the **Advanced > Networks** page, and click the **Create** button.

Specify the name and VLAN ID that you want to create for the VLAN network <small>(You can specify the same VLAN ID in different namespaces if you have [Rancher multi-tenancy](/rancher/virtualization-management/#multi-tenancy) configured)</small>.

   ![create-vlan-network.png](assets/create-network.png)

> ### Connect a VM to the Harvester management network

Reviewer 2: ditto

> ### Connect a VM to the Harvester management network

Author: ```suggestion
### Connect a VM to the `harvester-mgmt` network
```

> Once you finished the configuration in the previous section, the external switch will send out untagged network traffic to the management network. In Harvester, the untagged traffic is received in VLAN 1.

Author: ```suggestion
Once you finished the configuration in the previous section, the external switch will send out untagged network traffic to the subnet of the Harvester hosts. In Harvester, the untagged traffic is received in VLAN 1.
```

> Once you finished the configuration in the previous section, the external switch will send out untagged network traffic to the management network. In Harvester, the untagged traffic is received in VLAN 1.
> 
> If users need VM connects to VLAN 100, management network, users can create a VLAN Network in Harvester with VLAN ID 1 configuration instead of VLAN ID 100.

Reviewer: ```suggestion
Once you finished the configuration in the previous section, the external switch will send out untagged network traffic to the management network. In Harvester, the untagged traffic is received in VLAN 1.

Therefore, if you need VMs to connect to the VLAN ID 100, you need to create a VLAN ID 1 Network in Harvester.
```

The external switch will remove the VLAN 100 tag from the packet for egress and `harvester-br0` will add the VLAN 1 tag to the packet and treat it as VLAN 1 as shown in the following diagram:

   ![vlan-aware-native-vlan.png](assets/vlan-aware-native-vlan.png)

!!! warning
    Do not create a VLAN Network with VLAN 100 and associate any VM to it. The connectivity will not always be ensured and depends on the external switch behavior to add/remove VLAN tag from packets.

### Connect a VM to specific VLAN network

You need to create a VLAN Network with specific VLAN ID and associate the VM to that VLAN network. 

> Please refer to [this page](/networking/harvester-network/) for additional information on Harvester Networking.

Reviewer 3: ditto.
